### PR TITLE
Add functionality to rebalance an account

### PR DESCRIPTION
**Pull request by virgilbourassa**
Added functionality to rebalance an account based on proportional allotments to desired stocks.
Also addressed PyCharm IDE warnings and formatting.

### CODE REVIEW

File: schwab_api/schwab.py [hunk 1]
```diff
@@ -164,53 +180,412 @@ def trade(self, ticker, side, qty, account_id, dry_run=True):
...
+        return self.get_current_prices([ticker]).get(ticker)
+
+    def get_available_funds(self, account_id: int) -> float:
+        """Get cash funds available for purchases in the account
+        :param account_id:  int
+        """
+        accounts = self.get_account_info_v2()
+        return accounts[account_id]["available_cash"]
+
+    def rebalance(self, account_id: int, allocations: dict[str, float], set_aside: float = 0.0,
```
Comment (itsjafer) [inline schwab_api/schwab.py:243]: While I like the additional functionality that this PR introduces, I'm not sure whether a rebalance function should be provided by the API -- my preference would be that the API provides functions that allows clients to implement this rebalance function.

i.e. this API mirrors functionality provided by Schwab out-of-the-box and allows clients to implement more complex functions on their own

**Comment by itsjafer** [comment]
Additionally, I believe this PR renames certain methods, which introduces breaking changes in terms of backwards compatibility. 

I think we need strong justification before we take that step.

**Comment by virgilbourassa** [comment]
I'm happy to undo the renaming - I was trying to adhere to the recommended conventions that PyCharm keeps warning me about. Thanks for your comments and consideration.

File: schwab_api/schwab.py (hunk 1 above)
Comment (virgilbourassa) [reply schwab_api/schwab.py:243]: That's fair - I'm enjoying using this to automate my monthly rebalancing of about 10 accounts. As of today, 10/18/2024, however, it seems that Schwab has changed their login page so the authentication is no longer working.

Comment (virgilbourassa) [reply schwab_api/schwab.py:243]: Nevermind - I needed to catch up on your commits regarding stealth mode. Now it works again.

**Pull request closed without merging**